### PR TITLE
Deployment Script Hardhat

### PR DESCRIPTION
**Pull request by ironFe93**
## What

Recreates the deployment script using hardhat + ethers

## Why

upgrade

## Refs

https://rsklabs.atlassian.net/browse/PP-438

### CODE REVIEW

File: scripts/modules/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const { address: customDeployVerifierAddress } =
+    await customSmartWalletDeployVerifierF.deploy(
+      customSmartWalletFactoryAddress
+    );
+
+  const { address: customRelayVerifierAddress } = await relayVerifierF.deploy(
+    customSmartWalletFactoryAddress
+  );
+
+  const { address: utilTokenAddress } = await utilTokenF.deploy();
```
Comment (jurajpiar) [inline scripts/modules/deploy.ts:89]: I don't think we want to deploy the util token unless it's a testnet, right?

File: test/scripts/deploy.test.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        'CustomSmartWalletDeployVerifier',
+        'CustomSmartWalletRelayVerifier',
+        'UtilToken'
+      );
+    });
+
+    it('should deploy contracts with valid addresses', async function () {
+      const result = await deployContracts();
+      Object.values(result).forEach((value) => {
+        expect(value).to.eq(testAddress);
```
Comment (jurajpiar) [inline test/scripts/deploy.test.ts:49]:
```suggestion
        expect(value, value).to.eq(testAddress);
```
to show which one failed if any

**Review by jurajpiar — APPROVED** [review]
LG, just resolve all comments, please

**Review by kenny08gt — APPROVED** [review]
Working with [RSK Hop Release v4.1.0](https://github.com/rsksmart/rskj/releases/tag/HOP-4.1.0)

File: scripts/modules/deploy.ts (hunk 1 above)
Comment (ironFe93) [reply scripts/modules/deploy.ts]: I interpret this as "do not deploy UtilToken if network is mainnet". I put in this check and its corresponding test.

**Pull request merged**